script.js

```diff
@@ -57,49 +84,99 @@ const addCardContainer = () => {
-    if (inputField.value) {
-      if (e.key === "Enter") {
-        addListItem();
-      } else if (e.key === "Escape") {
-        inputField.value = "";
-      }
-      saveData();
-    };
-  });
 
   const listButton = document.createElement('button');
   listButton.setAttribute("class", "button");
   listButton.innerHTML = "Add";
-  listButton.addEventListener("click", () => {
-    if (inputField.value) {
-      addListItem();
-    };
-  });
 
   const listContainer = document.createElement('ul');
   listContainer.setAttribute("class", "list-container");
 
-  container.appendChild(cardContainer);
-
   cardContainer.append(deleteCardIcon, cardTitle, inputContainer, listContainer);
-  
+
   inputContainer.append(inputField, listButton);
 
-  listContainer.addEventListener("click", (e) => {
-    if (e.target.tagName.toLowerCase() === deleteItemTag) {
-      e.target.parentElement.remove();
-    } else if (e.target.tagName.toLowerCase() === inputItemTag) {
-      e.target.toggleAttribute('checked');
+  const todoListItemElems = todoListState
+    .map((todoListItemData) => renderTodoListItem(todoListItemData))
+    .map(({ elem }) => elem);
+
+  listContainer.append(...todoListItemElems);
+
+  const createNewTodoItem = () => {
+    if (!inputField.value) {
+      return;
+    };
+
+    const newTodoItemData = { text: inputField.value, isDone: false };
+    todoListState.push(newTodoItemData);
+
+    const { elem } = renderTodoListItem(newTodoItemData, () => {
+      const i = todoListState.indexOf(newTodoItemData);
+      todoListState.splice(i, 1);
+
+      notifyStateUpdated();
+    });
+
+    listContainer.append(elem);
+
+    inputField.value = '';
+
+    notifyStateUpdated();
+  };
+
+  inputField.addEventListener('keydown', (e) => {
+    if (e.key === "Enter") {
+      createNewTodoItem();
+    } 
+    
+    if (e.key === "Escape") {
+      inputField.value = "";
     };
-    saveData();
   });
+
+  listButton.addEventListener('click', () => createNewTodoItem());
+
+  deleteCardIcon.addEventListener('click', () => {
+    cardContainer.remove();
+    onDestroy();
+  });
+
+  return {
+    elem: cardContainer
+  };
 };
 
-addToListButton.addEventListener ("click", () => {
-  addCardContainer();
-});
+const renderTodoListItem = (todoListItemState, onDestroy) => {
+  const listItem = document.createElement('li');
+
+  const listItemInput = document.createElement('input');
+  listItemInput.setAttribute("type", "checkbox");
+
+  const listItemLabel = document.createElement('label');
+
+  const deleteItem = document.createElement('span');
+  deleteItem.innerHTML = "☒";
+
+  listItem.append(listItemInput, listItemLabel, deleteItem);
+
+  listItemLabel.innerHTML = todoListItemState.text;
+
+  listItemInput.addEventListener('click', () => {
+      listItemInput.toggleAttribute('checked', todoListItemState.isDone = true);
+      notifyStateUpdated();
+  });
+
+  deleteItem.addEventListener('click', (e) => {
+    e.target.parentElement.remove();
```

Comment:
This case it's better to address the `li` element through the variable `listItem`, so you don't need to use additional properties